### ISSUE TITLE
[Bug]: UI Shell tabbing hidden elements in the UI

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#14151]
### Package

@carbon/react

### Package version

1.32.0

### React version

18.2

### Description

In the `SideNav` component there are a few things to improve:

- Replicate the behaivor of the `Switcher`, which is **NOT** open the `SideNav` on focus, but keep closing when loses focus
- If the first fix is done, probably it won't be necessary the prop `onSideNavBlur`
- If the prop `onSideNavBlur` needs to be kept, it will be necessary to fix the conflict between it and the `isFixedNav `

Slack conversation: https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1688406261790019

### Reproduction/example

https://stackblitz.com/edit/github-mpwg6m?file=src%2FApp.jsx

### Steps to reproduce

Tab the page with the `SideNav` collapse when the prop `isFixedNav` is active and you will see that it is possible to tab the hidden items.

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
